sdk/core/azure-core/src/main/java/com/azure/core/util/ConfigurationSource.java

```diff
@@ -0,0 +1,34 @@
+package com.azure.core.util;
+
+import java.util.Map;
+
+public interface ConfigurationSource {
+    String getValue(String propertyName);
+
+    public static final ConfigurationSource ENV_VAR_SOURCE = new ConfigurationSource() {
+        @Override
+        public String getValue(String propertyName) {
+            return System.getenv(propertyName);
+        }
+    };
+
+    public static final ConfigurationSource SYSTEM_PROP_SOURCE = new ConfigurationSource() {
+        @Override
+        public String getValue(String propertyName) {
+            return System.getProperty(propertyName);
+        }
+    };
+
+    // TODO(configuration) should we ship one by default right away or leave it to users?
```

Comment:
I wouldn't eagerly ship a file based configuration source until we have more requests for it and are happy with the configuration naming convention. That said, we should definitely include this as a sample.